pkg/resource/cluster/hook.go

```diff
@@ -269,16 +278,20 @@ func (rm *resourceManager) customUpdate(
 		return returnClusterUpdating(updatedRes)
 	}
 	if delta.DifferentAt("Spec.UpgradePolicy") {
-		if err := rm.updateClusterUpgradePolicy(ctx, desired); err != nil {
-			awserr, ok := ackerr.AWSError(err)
-			// Check to see if we've raced an async update call and need to
-			// requeue
-			if ok && awserr.Code() == "ResourceInUseException" {
-				return nil, requeueAfterAsyncUpdate()
+		if updatedRes.ko.Spec.UpgradePolicy == nil {
+			updatedRes.ko.Spec.UpgradePolicy = latest.ko.Spec.UpgradePolicy.DeepCopy()
```

Comment:
This should never happen since you're setting it already in `preCompareUpgradePolicy`

Reply:
forgot about this one..thanks for catching it